src/components/ChatContent/Messages/BlockMenu/MessageContextMenu.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <vue-context ref="menu">
+    <ul>
+      <li
+        @click="goToUserProfile"
+        style="border-bottom: 1px solid black;"
+      >{{ message ? message.user.username : ''}}
+      </li>
+      <li @click="openReportModal({report: true, ignore: false})">Report User / Message</li>
+      <li @click="openReportModal({report: false, ignore: true})">Ignore User</li>
+    </ul>
+  </vue-context>
+</template>
+
+<script lang="ts">
+  import { Component, Prop } from 'vue-property-decorator';
+  import Vue from '@/types/vue';
+  import { VueContext } from 'vue-context';
+  import Message from '../../../../types/message';
+  import User from '@/types/user';
+
+  @Component({
+    components: {
+      VueContext,
+    },
+  })
+  export default class MessageContextMenu extends Vue {
+    $refs!: {
+      menu: HTMLFormElement;
+    };
+
+    message: Message | null = null;
+
+    open: boolean = false;
+
+    get user(): User {
+      return this.message!.user;
+    }
+
+    onCtxOpen() {
+      this.open = true;
+    }
+
+    onCtxClose() {
+      this.open = false;
+    }
+
+    created() {
+      this.$store.subscribe((mutation, state) => {
+        if (mutation.type === 'openContextMenu') {
```

Comment:
It seems that the reason I have originally made the context menu a global component is that it disappears alongside the button when I stop hovering over the message.
Do you think it would make more sense to send a boolean prop to the button component which says whether it should appear or not, and handling the visibility from inside, or to have the context menu be a child of the message? I'm in favor of the first option.

Reply:
I still think that it makes more sense from a conceptual standpoint for the menu to be a child of the message, triggered by the "button". When clicked, the button raises an event which will cause a prop whether the menu is shown or not to be flipped.

This is just a fancy button - it's scope of responsibility be just like any other button.

Reply:
But that context menu feels connected to the button. It's like how the menu in eterna is conceptually an extension of the menu button, at least in my opinion

Reply:
After a drawn out discussion with myself, I agree with you. Particularly persuaded by looking at Slack, where they have multiple buttons on a message, the menu does seem to be localized to the set of buttons, not the message. That said - I'd rename the component MessageActions or such.

As a general note: "connected"/"extension of" is not the same thing as "contained by"/"is a child of". With submenus, they are actually children - ie, the resources menu item "contains" a strategy guides menu item. In this case, I do think that it is a child - but not a child of the *button* - the button is actually just the anchor tag, hence why I suggest renaming this.

Reply:
How would you suggest naming it? I can't think of anything.

Reply:
What do you think about `MessageActions`?

Reply:
That's great